ocaml/xapi/xapi.ml

```diff
@@ -577,6 +598,7 @@ let maybe_set_fqdn_in_pool_conf () =
           match Gencertlib.Lib.fqdn_of_hostname master_hostname with
           | Some master_fqdn ->
               if persisted_master_address <> master_fqdn then (
+                assert_can_ping master_fqdn ;
```

Comment:
I think it would be preferable to investigate what race condition is being solved by waiting 5 seconds, and try to see what we can do about that other than waiting some seconds as it seems fragile.

Reply:
I don't think we have a race condition necessarily, we genuinely may not be able to reach whatever FQDN we have calculated the master to be at (in which case it would be a bad idea to go ahead and change the pool config).

Part of me would like to do the pool config changes only when TLS verification is enabled, since at that point we can expect all the hosts to be online, and we can give the user an error message if the FQDNs are wrong. Admittedly we could propagate an error message anyway using the host health check we added: https://github.com/xapi-project/xen-api/commit/fb61b55e51f9d7a0199b588d1df432321ca3a8a5